app/scripts/views/form.js

```diff
@@ -465,18 +465,30 @@ define(function (require, exports, module) {
       return message;
     },
 
+    reevaluateSubmissionEnabling: function (self){
```

Comment:
not sure if you need this function or we can just keep it all in `beforeSubmit` and just better unit test that function